lntest/itest/lnd_recovery_test.go

```diff
@@ -126,27 +129,30 @@ func testOnchainFundRecovery(net *lntest.NetworkHarness, t *harnessTest) {
 	carol, mnemonic, _, err := net.NewNodeWithSeed(
 		"Carol", nil, password, false,
 	)
-	if err != nil {
-		t.Fatalf("unable to create node with seed; %v", err)
-	}
+	require.NoError(t.t, err)
 	shutdownAndAssert(net, t, carol)
 
+	// As long as the mnemonic is non-nil and the extended key is empty, the
+	// closure below will always restore the node from the seed. The tests
+	// need to manually overwrite this value to change that behavior.
+	rootKey := ""
+
 	// Create a closure for testing the recovery of Carol's wallet. This
 	// method takes the expected value of Carol's balance when using the
 	// given recovery window. Additionally, the caller can specify an action
 	// to perform on the restored node before the node is shutdown.
 	restoreCheckBalance := func(expAmount int64, expectedNumUTXOs uint32,
 		recoveryWindow int32, fn func(*lntest.HarnessNode)) {
 
+		t.t.Helper()
+
 		// Restore Carol, passing in the password, mnemonic, and
 		// desired recovery window.
 		node, err := net.RestoreNodeWithSeed(
-			"Carol", nil, password, mnemonic, "", recoveryWindow,
-			nil,
+			"Carol", nil, password, mnemonic, rootKey,
```

Comment:
nit: could create `rootkey` in previous commit so we don't have to touch this line twice in a row

Reply:
Ah yes, definitely!